data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java

```diff
@@ -455,7 +459,7 @@ private <T, R> Stream<R> findStream(@NonNull PreparedQuery<T, R> pq, Connection
                     QueryResultInfo queryResultInfo = preparedQuery.getQueryResultInfo();
                     if (queryResultInfo != null && queryResultInfo.getType() == QueryResult.Type.JSON) {
```

Comment:
Personally, I would try to extract this code to some factory.

Reply:
Where based on the dialect and `PersistentProperty` we can decide which impl. to choose.
Also we can have some kind of registry which would allow as to add a possible mapper only if Oracle and Serde OSON is on the classpath.

Reply:
I have added `JsonColumnReader` so reading JSON is now on column level and if there is custom json column reader for the dialect (in this case Oracle and oson mapper) then it will be used.
It is a bit concerning, I don't know if JSON field from Oracle will always return `OracleJsonObject` and not sure where we can/want to configure that.